### PR TITLE
Migrate setup to the new database form

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26178
Blocked by https://github.com/metabase/metabase/pull/26819

How to test:
- Run a fresh metabase instance
- Make sure it is possible to connect your db during setup

Screenshot 2022-11-30 at 13 51 38

### CODE REVIEW

**Comment by kulyk** [comment]
@ranquild once I pick the Postgres, the "Display name" field is prefilled with a `"null"` string. On `master` it's empty

CleanShot 2022-11-30 at 13 53 35@2x

File: frontend/src/metabase/databases/utils/schema.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export const getValidationSchema = (
 
   return Yup.object({
     engine: Yup.string().default(engineKey).required(Errors.required),
-    name: Yup.string().nullable().default(null).required(Errors.required),
+    name: Yup.string().default("null").required(Errors.required),
```
Comment (kulyk) [inline frontend/src/metabase/databases/utils/schema.ts:18]: Not sure it leads to expected behavior

**Review by kulyk — APPROVED** [review]
Awesome!

File: frontend/src/metabase/databases/utils/schema.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/databases/utils/schema.ts:18]: It's not. It's probably the funniest typo I've ever made :)

Comment (ranquild) [reply frontend/src/metabase/databases/utils/schema.ts]: Fixed

**Pull request merged**